doc/perception_configuration/perception_configuration_tutorial.rst

```diff
@@ -102,3 +116,63 @@ MoveIt! uses an octree-based framework to represent the world around it. The *Oc
  * *octomap_frame*: specifies the coordinate frame in which this representation will be stored. If you are working with a mobile robot, this frame should be a fixed frame in the world.
  * *octomap_resolution*: specifies the resolution at which this representation is maintained (in meters).
  * *max_range*: specifies the maximum range value to be applied for any sensor input to this node.
+
+Obstacle Avoidance
+==================
+
+If you set the initial and the final location of the robot in a way that there is no straight path between them, then the planner will automatically avoid the octomap and plan around it.
+
+.. image:: obstacle_avoidance.gif
+   :width: 700px
+
+Running the interface
+---------------------
+Roslaunch the launch file to run the code directly from moveit_tutorials: ::
+
+ roslaunch moveit_tutorials obstacle_avoidance_demo.launch
+
+you should see something like the image shown at the beginning of this tutorial.
+
+You can test obstacle avoidance for yourself by setting the start and goal state manually and then planning. To learn how to do that look at `MoveIt! Quickstart in RViz <../quickstart_in_rviz/quickstart_in_rviz_tutorial.html>`_
+
+Preprocessing The 3D Data
+=========================
+
+In this section, we will demonstrate an example on preprocessing the 3D data and integrating it with the MoveIt! perception pipeline.
+We will be working with point clouds but it can be implimented simmilarly with depth maps.
+
+In order to use the preprocessed pointcloud/depthmap with MoveIt! all you need to do is modify the ``point_cloud_topic``/``image_topic`` parameter to the topic where the processed pointcloud/depthmap is published.
+
+Getting Started
+---------------
+In addition to the steps given in `Getting Started <../getting_started/getting_started.html>`_, you will also need to install the ``perception_pcl`` package: ::
+
+ sudo apt install ros-kinetic-perception-pcl
+
+Running the Code
+----------------
+Edit the ``point_cloud_topic`` in the yaml configuration file created `here <./perception_configuration_tutorial.html#yaml-configuration-file-point-cloud>`_ ::
+
+ sensors:
+   - sensor_plugin: occupancy_map_monitor/PointCloudOctomapUpdater
+     point_cloud_topic: /processed_points
+     max_range: 5.0
+     point_subsample: 1
+     padding_offset: 0.1
+     padding_scale: 1.0
+     filtered_cloud_topic: filtered_cloud
+
+Roslaunch the launch file to run the code directly from moveit_tutorials: ::
+
+ roslaunch moveit_tutorials preprocessing_3d_data_demo.launch
+
+You should now see something like this:
+
+.. image:: preprocess_3d_demo.png
+   :width: 700px
```

Comment:
This is not good enough.
The example should leave the point cloud that is used for the octomap as-is and spawn a (finite-length) cylinder `CollisionObject` that can be used as the object model to grasp.